### PR TITLE
addressing Tom's feedback reg data science methods - PART 2

### PR DESCRIPTION
**Pull request by larahabashy**

### CODE REVIEW

File: _build/html/_sources/capstone-report.md [hunk 1]
```diff
@@ -44,26 +44,31 @@ name: counts_df
 ---
 The total number of positive and negative examples in our dataset.
 ```
-
-Before beginning any model training, the data was split into train, validation and test splits of 70%, 15% and 15% respectively. Here, we define baseline as a model that only predicts negative and has 62% accuracy. Our goal for this project was to develop a model that performed better than baseline. 
- 
 ## 3. Literature Review
 
-Throughout the project, the team has consulted literature to gain insight and direction on current practices relevant to our problem and our dataset to guide and validate our decision-making. Please see [Appendix A for our literature review](../lit_review.md).
+Throughout the project, the team has consulted literature to gain insight and direction on current practices relevant to our problem and our dataset to guide and validate our decision-making. Please see [Appendix A for our complete literature review](../lit_review.md).
```
Comment (jraza19) [inline _build/html/_sources/capstone-report.md]: consider adding back the entire section of literature review.

File: _build/html/_sources/capstone-report.md [hunk 2]
```diff
@@ -44,26 +44,31 @@ name: counts_df
...
-Before beginning any model training, the data was split into train, validation and test splits of 70%, 15% and 15% respectively. Here, we define baseline as a model that only predicts negative and has 62% accuracy. Our goal for this project was to develop a model that performed better than baseline. 
- 
 ## 3. Literature Review
 
-Throughout the project, the team has consulted literature to gain insight and direction on current practices relevant to our problem and our dataset to guide and validate our decision-making. Please see [Appendix A for our literature review](../lit_review.md).
+Throughout the project, the team has consulted literature to gain insight and direction on current practices relevant to our problem and our dataset to guide and validate our decision-making. Please see [Appendix A for our complete literature review](../lit_review.md). 
 
 ## 4. Data Science Methods 
 
+The general data science workflow for this project is shown in Figure 3. We discuss each step of the workflow in more detail in the sub-sections below.
```
Comment (jraza19) [inline _build/html/_sources/capstone-report.md]: replace figure 3 with {numref}`workflow`

File: _build/html/_sources/capstone-report.md [hunk 3]
```diff
@@ -75,7 +80,7 @@ Final image transformations included random vertical and horizontal flipping and
...
+The augmented data is then used to train a CNN model using transfer learning, a technique utlizing pre-trianed models which allows for training with comparatively little data. Based on our literature review, the transfer learning architectures we chose to investigate were the following: VGG16, ResNet50, DenseNet169 and InceptionV3. Each of the models were incorporated with our small dataset, trained in separate experiments utilizing techniques to optimize the parameters of the model to maximize its ability to learn. To compare the performance of the different architectures, the team considered the accuracy and recall scores of the models when tested on our test set. When comparing the relative scores by looking at {numref}`acc_chart`, DenseNet outperforms the other architectures.
```
Comment (jraza19) [inline _build/html/_sources/capstone-report.md]: spelling mistake: 'utilizing', 'pre-trained'

Suggestion: "a technique utilizing pre-trained models on thousands of images and then training them on similar but smaller datasets" (may not be the best but I wanted to mention the thousands of images part somehow)

Suggestion: "When comparing the relative scores of accuracy shown in {numref}`acc_chart` and recall shown in {numref}`recall_chart`, DenseNet outperforms the other architectures. 

Do we want to mention other literature that also had success with DenseNet?

File: _build/html/_sources/capstone-report.md [hunk 4]
```diff
@@ -104,12 +107,9 @@ name: true_pos_all_wrong
 ---
 A manual inspection of tricky examples was conducted. Above, we have a true positive and although, all model are struggling, DenseNet is the least confident in its wrong prediction.  
 ```
+In an effort to make the model more generalizable, we considered various techniques such as implemeting dropout layers in our CNN structure and taking an ensemble approach. Further, the medical nature of this project suggests that reducing false negatives (optimizing recall) may be more important than maximizing the overall accuracy on postive and negative predictions. To find more about these explorations, see [Appendix B - Documentation](../documentation.md).
```
Comment (jraza19) [inline _build/html/_sources/capstone-report.md]: Spelling: "implementing"

Suggestion:
"In an effort to make the model more generalizable, we considered various techniques such as implementing dropout layers in our CNN structure, batch normalization and taking an ensemble approach. Furthermore, when identifying liphypertrophic masses, it is more detrimental for a true positive (areas where lipohypertrophy is present) to be falsely labelled as a negative. This is formally known as a false negative and thus, we also attempted to optimize recall, a score where a higher value indicates fewer false negatives. To find out more about these explorations and their outcomes, see [Appendix B - Documentation](../documentation.md)."

File: _build/html/_sources/capstone-report.md [hunk 5]
```diff
@@ -104,12 +107,9 @@ name: true_pos_all_wrong
...
-
-Furthermore, as our capstone partner was far more interested in reducing false negatives, areas where insulin should not be injected, we investigated various techniques to optimize recall. Positive weight (pos_weight) is one argument of the loss function used in our CNN model which, if greater than 1, prioritizes the positive examples more such that there is a heavier penalization (loss) on labelling the positive lipohypertrophy examples incorrectly. However, this method was not implemented in our final model as it proved to be unstable. After conducting a few experiments, we found high variance in the [results](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/pos-weight-exploration.ipynb). 
-
-Finally, the choice of our optimal model as DenseNet was further motivated by its size and compatibility with our data product, further discussed in the data product section.
+Finally, the choice of our optimal model as DenseNet was further motivated by its relaively small computational size as seen in the table below, and its compatibility with our data product, further discussed in the data product section.
```
Comment (jraza19) [inline _build/html/_sources/capstone-report.md]: Replace "the table" with {numref}`size_df`

Replace "the data product section" with {ref}`data-product`

File: _build/html/_sources/capstone-report.md [hunk 6]
```diff
@@ -121,7 +121,7 @@ Comparison of the file size of the different models revealed that DenseNet was t
...
+Our next objective was to implement object detection into our pipeline, giving us the ability to identify the exact location of lipohypertrophy on unseen test images. To implement object detection using a popular framework called YOLO {cite:p}`glenn_jocher_2020_4154370`, the team created bounding boxes around the location of the lipohypertrophy masses in the positive training images using the annotated ultrasound images as a guide. Next, using the YOLOv5 framework, the YOLOv5m model was trained for 200 epochs with an image size of 320 and a batch size of 8. The team experimented with different training parameters to find that the aforementioned training parameters produce optimal results. Below is a sample of those results identifying the exact location of lipohypertrophy with great confidence.
```
Comment (jraza19) [inline _build/html/_sources/capstone-report.md]: the image size was 300!

File: _build/html/_sources/documentation.md [hunk 7]
```diff
@@ -58,11 +60,13 @@ The model variants that were considered are documented below, along with some no
...
+In an effort to reduce the generalization error and high confidence in misclassifications, we considered the implementation of dropout layers within the model's architecture. Dropout layers drop nodes in the network such that the model learns more robustly and the validation performance is improved. This is due to the fact that a dropout layer with random probability equal to the dropout rate, drop nodes from a network between layers. However, the dropout layers experiment with varying dropout rates did not show any success for our dataset. Although the dropout layers did manage to reduce misclassifications in the models, the overall reduced accuracy was not remarkable enough to implement those layers in our optimal model. To see this exploration, click [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/densemodels-ax-dropout-layers.ipynb).
```
Comment (jraza19) [inline _build/html/_sources/documentation.md]: We repeat "drop nodes.." twice. maybe we can delete "This is due to the fact..."

**Review by jraza19 — COMMENTED** [review]
@larahabashy awesome work! Just a few comments.

File: _build/html/_sources/capstone-report.md (hunk 4 above)
Comment (larahabashy) [reply _build/html/_sources/capstone-report.md]: Great suggestion, thank you!

File: _build/html/_sources/capstone-report.md (hunk 6 above)
Comment (larahabashy) [reply _build/html/_sources/capstone-report.md]: oops typo!
